reliability-engineering/terraform/modules/concourse-monitoring/files/dashboards/concourse-slo.json

```
@@ -0,0 +1,383 @@
+          "dimensions": {},
+          "expression": "1 - status5xx / (status5xx + status2xx)",
+          "hide": false,
+          "id": "errorRatio",
+          "matchExact": true,
+          "metricName": "",
+          "namespace": "",
+          "period": "",
+          "refId": "C",
+          "region": "default",
+          "statistics": [
+            "Average"
+          ]
+        }
+      ],
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "Availability",
+      "type": "stat"
+    },
+    {
+      "content": "\n# Concourse Availability\n\nWe monitor the ratio of successful vs \nfailure HTTP status\ncodes from the load balancer that sits\nin front of the concourse-web nodes.\n\nThis is a good proxy for if the concourse\nUI/API is up and available for use.\n\n**We aim for 99.9% of requests to\nsucceed over a 7day period.**\n\n\n\n\n",
+      "datasource": null,
+      "fieldConfig": {
+        "defaults": {
+          "custom": {}
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 9,
+        "x": 6,
+        "y": 0
+      },
+      "id": 12,
+      "mode": "markdown",
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "",
+      "transparent": true,
+      "type": "text"
+    },
+    {
+      "datasource": null,
+      "description": "Percentage of requests faster than 250ms",
+      "fieldConfig": {
+        "defaults": {
+          "custom": {},
+          "decimals": 2,
+          "mappings": [
+            {
+              "from": "",
+              "id": 0,
+              "operator": "",
+              "text": "",
+              "to": "",
+              "type": 1,
+              "value": ""
+            }
+          ],
+          "noValue": "0",
+          "thresholds": {
+            "mode": "percentage",
+            "steps": [
+              {
+                "color": "dark-red",
+                "value": null
+              },
+              {
+                "color": "dark-yellow",
+                "value": 95
+              },
+              {
+                "color": "dark-green",
+                "value": 99
+              }
+            ]
+          },
+          "unit": "percentunit"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 6,
+        "x": 0,
+        "y": 8
+      },
+      "id": 2,
+      "options": {
+        "colorMode": "value",
+        "graphMode": "area",
+        "justifyMode": "auto",
+        "orientation": "auto",
+        "reduceOptions": {
+          "calcs": [
+            "mean"
+          ],
+          "fields": "",
+          "values": false
+        }
+      },
+      "pluginVersion": "7.0.5",
+      "targets": [
+        {
+          "expr": "sum(rate(concourse_http_responses_duration_seconds_bucket{le=\"0.25\",status!=\"500\"}[5m]))\n/\nsum(rate(concourse_http_responses_duration_seconds_count[5m]))\n\n",
+          "interval": "",
```

```suggestion
          "interval": "5m",
```
Since we're aggregating over 5-minute buckets, we shouldn't request data at a more fine-grained level than this, both for performance (we fetch less data) and correctness (we don't double-count the same request in two different buckets).